src/main/java/com/buyanywhere/productcatalog/controllers/v1/CategoryController.java

```diff
@@ -1 +1,33 @@
+package com.buyanywhere.productcatalog.controllers.v1;
 
+import com.buyanywhere.productcatalog.exceptions.CategoryNotFoundException;
+import org.springframework.web.bind.annotation.*;
+import com.buyanywhere.productcatalog.models.Category;
+import com.buyanywhere.productcatalog.repositories.CategoryRepository;
+
+@RestController
+@RequestMapping("/v1/category")
+
+public class CategoryController {
+
+    private CategoryRepository repository;
+
+    public CategoryController(CategoryRepository repository) {
+        this.repository = repository;
+    }
+
+    @RequestMapping(method = RequestMethod.POST, value = "/create-new")
+    public Category post(@RequestBody Category category) {
+        return repository.save(category);
+    }
+
+    @RequestMapping(method = RequestMethod.GET, value = "/get/{id}")
+    @ResponseBody
+    public Category get(@PathVariable("id") long id) throws CategoryNotFoundException {
+        if (repository.findById(id).isPresent() && !repository.findById(id).get().isDeleted()) {
```

Comment:
You could revert this if. This will allow you to assert that the category searched for is not present or is deleted, in which case you throw the corresponding exception. You should allways try to break as soon as possible. Meaning, you know that if the category is deleted or non existent then you should throw an exception. This allows for shorter processing and cleaner code. Let me know if you need clarification about this

Reply:
This can be refactored as:
`if(!repository.findById(id).isPresent() || repository.findById(id).get.isDeleted()) {`
`   throw new CategoryNotFoundException(id);`
`}`

`return repository.findById(id).get();`

Reply:
The refactor for this part of the code was appplied.  As you mentioned, it makes the code clearer. 
I will ask you for clarification about processing.
@maximilianotopp